packages/web3/src/TestGatingUtils.ts

```diff
@@ -0,0 +1,141 @@
+import { keccak256, parseEther } from 'viem/utils'
+import { foundry } from 'viem/chains'
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts'
+import { createTestClient, http, publicActions, walletActions } from 'viem'
+
+import type { Abi } from 'abitype'
+
+import { Address } from './ContractTypes'
+
+import { dlogger } from '@river-build/dlog'
+
+const logger = dlogger('csb:TestGatingUtils')
+
+export class Mutex {
+    queue: ((value: void | PromiseLike<void>) => void)[]
+    locked: boolean
+    constructor() {
+        this.queue = []
+        this.locked = false
+    }
+
+    lock() {
+        if (!this.locked) {
+            this.locked = true
+            return Promise.resolve()
+        }
+
+        let unlockNext: (value: void | PromiseLike<void>) => void
+
+        const promise = new Promise<void>((resolve) => {
+            unlockNext = resolve
+        })
+
+        this.queue.push(unlockNext!)
+
+        return promise
+    }
+
+    unlock() {
+        if (this.queue.length > 0) {
+            const unlockNext = this.queue.shift()
+            unlockNext?.()
+        } else {
+            this.locked = false
+        }
+    }
+}
+
+export function toEIP55Address(address: Address): Address {
+    const addressHash = keccak256(address.substring(2).toLowerCase() as Address)
+    let checksumAddress = '0x'
+
+    for (let i = 2; i < address.length; i++) {
+        if (parseInt(addressHash[i], 16) >= 8) {
+            checksumAddress += address[i].toUpperCase()
+        } else {
+            checksumAddress += address[i].toLowerCase()
+        }
+    }
+
+    return checksumAddress as Address
+}
+
+export function isEIP55Address(address: Address): boolean {
+    return address === toEIP55Address(address)
+}
+
+export function isHexString(value: unknown): value is Address {
+    // Check if the value is undefined first
+    if (value === undefined) {
+        return false
+    }
+    return typeof value === 'string' && /^0x[0-9a-fA-F]+$/.test(value)
+}
+
+export async function deployContract(
+    contractName: string,
+    abi: Abi,
+    bytecode: Address, // bytecode is a hex string
+    constructorArgs: any[] = [],
+): Promise<Address> {
+    let retryCount = 0
+    let lastError: unknown
+    while (retryCount++ < 5) {
+        try {
```

Comment:
there is a ton of copy pasta here. I'm 99.9% sure we fixed the errors by using the `const privateKey = generatePrivateKey()` here (how can nonce be too low if we create a new account for every call?).

Can you remove all the mutexes and retries and try catches and just see if this works without all the extra stuff? It it gets unstable make a bug and assign it to me and I'll put it all back.

Reply:
Oh wait! you refactored the other stuff into this one. Sorry i missed that. Lets leave it!